github_discovery.py

```diff
@@ -503,526 +488,476 @@ def process_repo(**component):
     except Exception:
       log.debug('No hmpps orb version found')
 
-  # Helm charts
-  helm_chart = (
-    get_file_yaml(repo, f'{helm_dir}/{c_name}/Chart.yaml')
-    or get_file_yaml(repo, f'{helm_dir}/Chart.yaml')
-    or {}
-  )
-  if 'dependencies' in helm_chart:
-    helm_dep_versions = {}
-    for item in helm_chart['dependencies']:
-      helm_dep_versions.update({item['name']: item['version']})
-    versions_data.update({'helm_dependencies': helm_dep_versions})
-
-  helm_environments = []
   try:
     helm_deploy = repo.get_contents(helm_dir, default_branch.commit.sha)
   except Exception as e:
     helm_deploy = False
     log.debug(f'helm_deploy folder: {e}')
+  try:
+    repo_env_list = repo.get_environments()
+  except Exception as e:
+    log.debug(f'helm_deploy folder: {e}')
+  existing_envs = component['attributes']['environments']
+  stored_commit_data = component['attributes']['latest_commit']
+
+  # Compare the current commit SHA with the stored SHA
+  if stored_commit_data and stored_commit_data.get('sha') != current_commit['sha'] or (not existing_envs and repo_env_list.totalCount > 0):
+    log.info(f'Repo {github_repo} has changed since last github discovery update.')
+    # Helm charts
+    if helm_deploy:
+      helm_commits = repo.get_commits(path=helm_dir)
```

Comment:
is there a reason it's just helm related stuff we're protecting with the commit id check? 
Is there other stuff - circle ci, docker, build.gradle/package.json that would all benefit from the checking latest commit?